### PR TITLE
compositing: Send entire scene's scroll offsets when sending WebRender display lists

### LINKED ISSUE #31807: Scroll offsets are sometimes lost when updating WebRender display list
To reproduce:

1. Create test.html as below (and iframe.html with enough text to scroll)
2. Scroll the iframe down to some non-zero y
3. Observe that the iframe appears to scroll back to the top

```html
<!doctype html><meta charset="utf-8">
<style>
    div {
        transform: translateY(7em) rotate(15deg);
    }
    .toggle {
        transform: translateY(7em) rotate(30deg);
    }
</style>
<div><iframe src="iframe.html"></iframe></div>
<script>
    setInterval(() => {
        document.querySelector("div").classList.toggle("toggle");
    }, 500);
</script>
```

### PR DESCRIPTION
**Pull request by mrobinson**
WebRender does not preserve spatial tree offsets when updating the
spatial tree. Updating the spatial tree of a pipeline can also
update the spatial tree of child pipelines. This change ensures that
WebRender always gets the scroll offsets of the entire scene when
modifying display lists in a way that may rebuild the spatial tree.

Fixes #31807.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #31807.
- [x] There are tests for these changes

### CODE REVIEW

**Comment by mrobinson** [comment]
Hey @delan, just let me know if you'd like me to hold off on this one until you've landed #31417.

**Comment by delan** [comment]
> Hey @delan, just let me know if you'd like me to hold off on this one until you've landed #31417.

Go ahead, I probably won’t be able to land my patch until Monday anyway.

**Review by delan — COMMENTED** [review]
The impl looks good, but can we have an automated test for this?

**Comment by mrobinson** [comment]
> The impl looks good, but can we have an automated test for this?

At first I didn't think so, but after a bit of pondering it turns out that there is a way to test this. I've added that test to the latest version of the branch.

**Review by delan — APPROVED** [review]

**Pull request merged**
